account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,192 @@
+   * Retrieves the current user information.
+   *
+   * @returns {Promise<User>} A promise that resolves to the user object
+   */
+  public whoami = async (): Promise<User> => {
+    if (this.user) {
+      return this.user;
+    }
+
+    const stampedRequest = await this.turnkeyClient.stampGetWhoami({
+      organizationId: this.orgId,
+    });
+
+    const user = await this.alchemyClient.whoami(stampedRequest);
+    this.user = user;
+    return user;
+  };
+
+  /**
+   * Generates a stamped whoami request. Using this stamp is the most
+   * trusted way to get the user information since a stamp can only
+   * belong to the user who created it.
+   *
+   * @returns {Promise<TSignedRequest>} a promise that resolves to the "whoami" information for the logged in user
+   * @throws {Error} if no organization ID is provided
+   */
+  public stampWhoami = async (): Promise<TSignedRequest> => {
+    const user = await this.whoami();
+
+    return this.turnkeyClient.stampGetWhoami({
+      organizationId: user.orgId,
+    });
+  };
+
+  /**
+   * Signs a raw message
+   *
+   * @param {Hex} msg The message to be signed
+   * @param {SignRawMessageMode} mode The signing mode (default is "ETHEREUM")
+   * @returns {Promise<Hex>} A promise that resolves to the signed message
+   */
+  public signRawMessage = async (
+    msg: Hex,
+    mode: SignRawMessageMode = "ETHEREUM"
+  ): Promise<Hex> => {
+    const user = await this.whoami();
+
+    const stampedRequest = await this.turnkeyClient.stampSignRawPayload(
+      buildStampedSignatureRequestBody(user, msg, mode)
+    );
+
+    const { signature } = await this.alchemyClient.signPayload(stampedRequest);
+
+    return signature;
+  };
+}
+
+/**
+ * AlchemyApiKeySigner is a SmartAccountSigner that uses an ApiKeySignerClient to sign messages.
+ * Primarily intended to be used server-side.
+ *
+ * @example
+ * ```ts
+ *  const signer = new AlchemyApiKeySigner({
+ *   apiKey: {
+ *     privateKey: "private-api-key",
+ *     publicKey: "public-api-key",
+ *   },
+ *   connection: {
+ *     apiKey: "alchemy-api-key",
+ *   },
+ *   orgId: "user-org-id",
+ * });
+ *
+ * const account = await createModularAccountV2({
+ *   transport,
+ *   signer,
+ *   chain,
+ * });
+ *
+ * const client = createAlchemySmartAccountClient({
+ *   account,
+ *   transport,
+ *   chain,
+ * });
+ * ```
+ */
+export class AlchemyApiKeySigner implements SmartAccountSigner {
+  signerType = "alchemy-api-key-signer";
+  inner: ApiKeySignerClient;
+
+  /**
+   * Constructs a new instance of the AlchemyApiKeySigner.
+   *
+   * @param {ApiKeySignerClientParams} params The parameters for the client, including the API key and orgId
+   */
+  constructor(params: ApiKeySignerClientParams) {
+    this.inner = new ApiKeySignerClient(params);
+  }
+
+  /**
+   * Retrieves the current user's address.
+   *
+   * @returns {Promise<Address>} A promise that resolves to the user's address
+   */
+  getAddress = async (): Promise<Address> => {
```

Comment:
I'd suggest marking all these methods from the interface as `override`, to make it more obvious to the reader and also produce more helpful errors if the interface isn't correctly implemented.